### PR TITLE
Beefing up the tests for AMQP stuff in the guest to look for FD leaks.

### PR DESCRIPTION
**Pull request by TimSimpsonR**

### CODE REVIEW

**Comment by cp16net** [comment]
+1

File: tests/integration/tests/guest/amqp_restarts.py [hunk 1]
```diff
@@ -229,6 +242,12 @@ def send_message(self):
         matches = re.search("(\\w+)\\.(\\w+)\\.(\\w+)\\.(\\w+)", version)
         assert_true(matches is not None)
 
+    def _check_for_fd_growth(self):
+        actual_fdc = self.agent.get_fd_count()
+        if self.original_fdsize <  actual_fdc:
+            fail("Oh no, fdsize rose from %d to %d!"
```
Comment (ed-) [inline tests/integration/tests/guest/amqp_restarts.py:248]: <pre>UH - OH!
THE FDSIZE HAVE
STARTED TO MOVE!</pre>

**Comment by ed-** [comment]
+1

**Comment by jcru** [comment]
+1

**Comment by imsplitbit** [comment]
+1

**Pull request merged**